diff --git a/arch/x86/kernel/tsc.c b/arch/x86/kernel/tsc.c
index a1e641d..503a421 100644
--- a/arch/x86/kernel/tsc.c
+++ b/arch/x86/kernel/tsc.c
@@ -344,8 +344,10 @@ static int __init tsc_setup(char *str)
 		mark_tsc_unstable("boot parameter");
 	if (!strcmp(str, "nowatchdog"))
 		tsc_watchdog = TSC_WATCHDOG_OFF;
-	if (!strcmp(str, "directsync"))
+	if (!strcmp(str, "directsync")) {
 		tsc_allow_direct_sync = 1;
+		tsc_watchdog = TSC_WATCHDOG_OFF;
+	}
 	if (!strcmp(str, "recalibrate"))
 		tsc_force_recalibrate = 1;
 	if (!strcmp(str, "watchdog"))
